src/sa/sa.c

```diff
@@ -591,3 +591,42 @@ bool sa_is_any(const struct sa *sa)
 		return false;
 	}
 }
+
+
+#ifdef HAVE_INET6
+void sa_set_scopeid(struct sa *sa, int scopeid)
+{
+	if (!sa)
+		return;
+
+	sa->u.in6.sin6_scope_id = scopeid;
```

Comment:
the AF type must be checked here.

if the AF is not INET6, it will access the wrong element in the union